zbus_macros/src/iface.rs

```diff
@@ -1048,12 +1067,23 @@ fn get_args_from_inputs(
         }
 
         let args_from_msg = match method_type {
-            MethodType::Property(_) => quote! {
+            MethodType::Property(PropertyType::Getter) => quote! {
+                #server_arg_decl
+
+                #conn_arg_decl
+
+                #header_arg_decl
+            },
+            MethodType::Property(PropertyType::Setter) => quote! {
+                let hdr = header.as_ref().unwrap();
+
                 #server_arg_decl
 
                 #conn_arg_decl
 
                 #header_arg_decl
+
+                #signal_emitter_arg_decl
```

Comment:
can we refactor to first generate the common bits here? I think best would be to just set `signal_emitter_arg_decl` to be `quote! {}` and similarly for the header init for the setter.

Reply:
Do you mean we should just have a `MethodType::Property(_)` case (so no longer two separate cases for getter or setters) and just have `signal_emitter_arg_decl` expand to nothing for the getter, and so on?

I will certainly explore that.

Reply:
> > Do you mean we should just have a `MethodType::Property(_)` case (so no longer two separate cases for getter or setters) and just have `signal_emitter_arg_decl` expand to nothing for the getter, and so on?
> 
> Yes and no. :) I meant that we can avoid some duplication by initializing `signal_emitter_arg_decl` and the header declaration differently:
> 
> ```rust
> let (hdr_init, signal_emitter_arg_decl) = match method_type {
>      ...
> }
> ```
> 
> I actually didn't realize that we're duplicating this in case of non-property as well so this refactor would be even more valuable than I realized. :)

Got it. I hope it's better now. Thanks!